### DISCUSSION TITLE
Don't know. Slurm-web don't work at all.

### DISCUSSION THREAD

**Discussion by BlackS52** [rackslab/Slurm-web#236] [Q&A]
Wanted to install it to SUSE. Had many problems with different /etc/passwd, users www_data, _apt and etc. Umounted
Found that the docker file tried to install SLURM verstion 15 instead current v22. Fixed it. 
Found munge problem when a spawned a new munge process each 1 sec that lead to crash your system. Fixed it in /etc/services/munge/run

But slurm-web still don't show anything in web browser. Only one button and nothing more. 

![image](https://user-images.githubusercontent.com/55309488/198671400-56385640-ef00-4a0a-9f75-a01115f2e35f.png)

Why ? Don't know. Looks like a project is died. Hope that I wrong.
The great idea but don't worked realisation. 
Sad.

**Comment by wangzhen2271** [comment]
I have the same problem.

**Comment by carlos-encs** [comment]
Same issue here.
Need some help.

**Comment by hellozzcy** [comment]
Same issue here.
Need some help.

**Comment by rezib** [comment]
The problem is probably due to a background HTTP request with an unexpected result. Can you please check the network requests in your browser development console and find out what is going wrong?

**Reply by hellozzcy** [reply]
I want to know where the URL and path in cluster.config.json are obtained from. I have deployed a local Slurm cluster, but I don't know what to fill in the address in cluster.config.json. Are there any default listening addresses? Please help answer this question. Thank you.

**Reply by hellozzcy** [reply]
![image](https://github.com/rackslab/slurm-web/assets/126752048/c21e871c-2f02-4c3e-98f3-ba51c5d5ffed)

**Reply by rezib** [reply]
Some documentation for this configuration file is available here:
http://rackslab.github.io/slurm-web/installation.html#clusters-clusters-config-json

There are two HTTP WSGI applications involved in Slurm-web : _confdashboard_ and _restapi_. The _restapi_ must be deployed on all clusters, the _confdashboard_ must be deployed on a central server with the HTML page and the JS code. In a configuration with one local Slurm cluster, all these components can be deployed on one server.

The first application _confdashboard_ basically provides `cluster.conf.json` which contains the list of Slurm clusters, with the URL to access their corresponding _restapi_ components. The URL value depends on the way the _restapi_ application is launched (eg. with apache/nginx or as standalone Flask application), its server and listening port.

**Reply by hellozzcy** [reply]
I am deploying slurm-web using Docker. How can I view the restapi configuration inside Docker? I didn't see any introduction on the official website. Also, the confdashboard you mentioned, how is it configured to read the local slurm cluster? I also didn't see any introduction in the official documentation. I would appreciate your answers, thank you very much.

**Reply by rezib** [reply]
Yes, the Docker container was an external contribution and it misses information in the existing documentation. The configuration file in the container is binded to the `conf\` subdirectory in the current working:

https://github.com/rackslab/slurm-web/blob/90463564375148b5c67e485006c6e2a4da5eb05e/docker/run/run.sh#L4

The default empty array in `clusters.conf.json` should be fine in this configuration, as it considers the _restapi_ component to be available under the `slurm-restapi` path on the same HTTP server. The problem you are facing is probably not caused by a misconfigured `clusters.conf.json` but more due to the _restapi_ encountering an undefined issue leading to HTTP/500 response codes. You should probably dig into the logs of the _restapi_ component to discover what is going wrong there.

**Reply by hellozzcy** [reply]
Is there any dependency between the authorized_origins configuration items in the cluster.conf.json and restapi.conf files, and I checked the logs, the command used is docker logs -f XXXX, I did not find any errors, I want to find inside docker How can I check the running status of these two applications, restapi and confdashboard, I want to check their configuration items, I really need your answer Thanks

**Reply by rezib** [reply]
I guess you can find some interesting stuff in `/var/log/apache` inside the container, by opening an interactive shell with docker.

**Comment by rezib** [comment]
This question was about deprecated Slurm-web 2, then I close the discussion.
